src/NUnitFramework/framework/Internal/Execution/WorkItemQueue.cs

```diff
@@ -117,22 +138,29 @@ public bool IsEmpty
         /// <param name="work">The WorkItem to process</param>
         public void Enqueue(WorkItem work)
         {
-            lock (_syncRoot)
+            do
```

Comment:
On the 4.0 and later builds, this code uses ManualResetEventSlim for the event.  That class also uses CAS and spin waiting internally to avoid kernel calls wherever possible.  I originally added that mono class to the pre 4.0 builds, but it requires a bunch of support classes as well, and it did not seem to worth doing so for those legacy builds.  I still could, however.  For instance, my CF build does use that class, since my CF environment has it already.